Make app look good

noahtallen opened the pull request:
### Test:
- run app
- test that every screen looks a lot better (e.g. splash screen, login, logo, buttons, colors, fab)
- make sure it don't crash

lukemeierj commented:
Looks really good!  

I think the bio text could use some padding
<img width="361" alt="Screen Shot 2019-03-21 at 12 55 18 PM" src="https://user-images.githubusercontent.com/7787605/54770258-3cb51e80-4bd9-11e9-8346-10e819264554.png">

Solve with `bioText.textContainerInset = UIEdgeInsets(top: 12, left: 7, bottom: 12, right: 7)`:
<img width="348" alt="Screen Shot 2019-03-21 at 12 59 04 PM" src="https://user-images.githubusercontent.com/7787605/54770257-3cb51e80-4bd9-11e9-85a0-95ced941bc77.png">
<img width="336" alt="Screen Shot 2019-03-21 at 1 00 53 PM" src="https://user-images.githubusercontent.com/7787605/54770361-6c642680-4bd9-11e9-8fba-6423a32ffcc3.png">
Works on 5s as well

---

 Login screen doesn't format well on the 5s.  FYI, but probably not a big deal
<img width="355" alt="Screen Shot 2019-03-21 at 1 00 34 PM" src="https://user-images.githubusercontent.com/7787605/54770450-a0d7e280-4bd9-11e9-8962-9de3bdac4c36.png">

Works OK on other devices like iPhone 8
<img width="388" alt="Screen Shot 2019-03-21 at 1 05 43 PM" src="https://user-images.githubusercontent.com/7787605/54770680-1a6fd080-4bda-11e9-9732-c4101ae2ce09.png">

lukemeierj commented:
What do you think about purple topics?  It would be a nice way of differentiating a `BuddyButton` from a non-actionable data item.  I think it might be less pretty, but I think we should find some way of distinguishing actionable buttons, and this could do for now.

<img width="340" alt="Screen Shot 2019-03-21 at 1 32 38 PM" src="https://user-images.githubusercontent.com/7787605/54772565-fca46a80-4bdd-11e9-90cd-f0f5d3d0f9e5.png">
<img width="357" alt="Screen Shot 2019-03-21 at 1 32 13 PM" src="https://user-images.githubusercontent.com/7787605/54772566-fca46a80-4bdd-11e9-8b76-27087f11a2d7.png">

lukemeierj commented on Buddies/Onboarding/SignUpInfoVC.swift:
```diff
@@ -63,6 +65,7 @@ class SignUpInfoVC: LoginBase, UIImagePickerControllerDelegate, UINavigationCont
         // Do any additional setup after loading the view.
         bioText.delegate = self
         bioText.textColor = UIColor.lightGray
+        bioText.layer.cornerRadius = 5
```
I think the bio-text field looks nicer with a little padding.  

`bioText.textContainerInset = UIEdgeInsets(top: 12, left: 7, bottom: 12, right: 7)`

lukemeierj commented:
I've been asking around about our logo material.  I'll put some of the feedback here.

<img width="246" alt="Screen Shot 2019-03-21 at 1 39 03 PM" src="https://user-images.githubusercontent.com/7787605/54772901-c61b1f80-4bde-11e9-934b-46e69178bced.png">

Some of the suggestions/notes I received:
* Consider avoiding the eyes going out past the smiley
* The colors don't fit great with smiley faces.  The milky white eyes aren't normal for a smiley drawing.  People mentioned something about a blue background being weird, and yellow fitting more with smileys.
* The eyes are too big
* I tried rounding the edges if the smile, and people liked the current square smile more.
* Looks kind of buggish
* Suggested using crescents instead of big moons
<img width="270" alt="Screen Shot 2019-03-21 at 1 44 21 PM" src="https://user-images.githubusercontent.com/7787605/54773211-7db03180-4bdf-11e9-9b5c-c01be24d32d4.png">

I know a lot you guys have gotten a lot of positive feedback.  I've mostly gotten indifference or critique.  I'm not sure that means we have to do anything about it now, but I think we should keep an open mind about it.

lukemeierj commented:
As for testing the PR, it tests super well! 🎉 

Can confirm that everything looks so much better and there are no extra bugs I can tell from this.  

I'm really excited to see this in master.  This is the point where the app is really becoming a thing, I'm really excited.  Thanks for putting your color senses to good use, @noahtallen  and @JakeThurman ! Looks great!

JakeThurman commented:
I agree with all of Luke's feedback. Although
- I literally don't care either way about the smiley face changes, either is fine.


Also The purple topics is a better choice for distinguishing - I agree. Although It's not my fav look, it seems worth it.

noahtallen commented:
Hm. Definitely agree about a lot of stuff here. My one thing is I don’t think the purple topics look good, which I think is the goal of the PR. I like that they’re more distinguished, but I think they look worse. Hm. Maybe we should find a different color?

On Mar 21, 2019, 3:33 PM -0400, Jake Thurman <notifications@github.com>, wrote:
> I agree with all of Luke's feedback. Although
>
> • I literally don't care either way about the smiley face changes, either is fine.
> • The purple topics is a better choice for distinguishing - I agree. Although It's not my fav look, it seems worth it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

JadaTBK commented:
I think the crescent eyes on the smiley are a lot better; regarding the topic colors vs. button color, i think having a more "purple" but still darker color for the button would work better than the current blue. I know luke mentioned people thinking yellow would be a more appropriate color related to a smiley face. Do we wanna think about shifting from purple to yellow?

noahtallen commented:
Jake and I did look into some Yellow colors, but I don't really like yellow as a primary color, personally. maybe as an alt?

JadaTBK commented:
Yeah i think yellow would be good as an alt, would we want to have like orange be the primary color? To keep with the color family.

noahtallen commented:
i personally really enjoy the deep purple as the primary color

lukemeierj commented:
I think a slightly less dark primary color could be good.  The contrast between the bright whites and the very dark blue looks a little unfinished to me.  The apps I use usually have primaries that contrast less with the base white, and then occasionally dark, high contrast controls.

noahtallen commented:
@lukemeierj @JakeThurman @gyurisic This could use another look. New changes:
1. Topics in View Activity look different and aren't buttons any more
2. Topic cards on the topic tab look different
3. Topics on pick topics for create activity look different
4. Tint color is now programmatically set globally for the app to the "Blue" color. This means that even things like the text cursor and action buttons on alerts show up in the right color. As a result, the navigation buttons are now "blue" rather than purple. I think it looks more consistent and better
6. Changed the destructive color back to red, so we either need to be ok with that or come up with something better than the maroon we had before
7. Switched smileys out to the crescent ones. 
8. Implemented the padding thing Luke suggested
9. Renamed the `ControlColors` class to `Theme` because I wanted to put the border radius constant in there


Todo(?):
- [ ] Better style for filter button on search bar? @lukemeierj this is something you have thoughts about; I don't think we reached any conclusions about it
